File: 2138. Divide a String Into Groups of Size k/divideString.js
/**
 * @param {string} s
 * @param {number} k
 * @param {character} fill
 * @return {string[]}
 */
var divideString = function (s, k, fill) {
    let res = []
    let len = s.length
    if (len % k !== 0) {
        let remain = k - (len % k)
        for (let i = 0; i < remain; i++) {
            s += fill
        }
    }

    for (let i = 0; i < len; i += k) {
        res.push(s.substr(i, k))
    }
    return res;
};

File: 2138. Divide a String Into Groups of Size k/divideString.cpp
#include <iostream>
#include <vector>

using namespace std;



class Solution {
    public:
        vector<string> divideString(string s, int k, char fill) {
            vector<string> res;
            int sz = s.size();
    
            if(sz % k != 0){
                int remain = k - (sz % k);
                for(int i=0 ; i<remain; i++){
                    s+= fill;
                }
            }
    
            for(int i=0 ; i<sz; i+=k){
                res.push_back(s.substr(i,k));
            }
    
            return res;
        }
    };